spec/lib/advocacy_campaign_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+describe AdvocacyCampaign do
```

Comment:
It's great to have tests around this behavior, especially since we don't expect the spreadsheet to change much over time.

If you ever end up cranking out another scraper, don't feel pressured to write tests.  Most of these scrapers are just quick and dirty scripts to pull in data.

For the CTAA project, we're big on tests for the API, but scrapers the bar is lower (mostly because we expect the sites we scrape to change without warning and we don't want spend lots of time updating broken tests that describe what a webpage used to look like).

Reply:
That all makes sense to me! Totally agreed on the approach of not testing pure website scraping because it’s so brittle. Letting good error handling inform of us of problems is great.

I wrote tests for this for basically two reasons:

1. It made it easier for me to validate as I was going along that it was largely correct, since dev testing in the Rails console is annoying
2. Like you said, I figured the spreadsheet is less likely to change so why not test it.